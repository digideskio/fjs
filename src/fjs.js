/**
 * Copyright 2004-present Facebook. All Rights Reserved.
 */

'use strict';

import type {Input, Output} from './types';

import applyRules from './applyRules';
import * as babel from 'babel-core';
import getGlobalContext from './getGlobalContext';
import getMiddleware from './getMiddleware';
import getPrinters from './getPrinters';
import getRules from './getRules';
import print from './print';

export default function fjs(input: Input): Output {
  const {ast} = babel.transform(input.code);
  const tokens = print(
    getPrinters(),
    getMiddleware(),
    getGlobalContext(ast),
    ast,
  );
  const code = applyRules(getRules(), tokens);
  return {code};
}
