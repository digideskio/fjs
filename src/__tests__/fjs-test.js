/**
 * Copyright 2004-present Facebook. All Rights Reserved.
 */

import fjs from '../fjs/';

it('lives!!', () => {
  expect(fjs()).toBe(1);
});
