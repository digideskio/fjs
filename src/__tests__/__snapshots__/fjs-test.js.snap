exports[`fjs changes-quotes 0`] = `
"let x = 'foobar';
x = 'foobar';
x = '\"';
x = '\"';
x = '\\'';
x = '\\'';
"
`;

exports[`fjs large-numbers 0`] = `
"100000000000000000000123456789;
"
`;

exports[`fjs simple-block-comment 0`] = `
"/*
 * comment
 */
1 + 1;
"
`;

exports[`fjs simple-comment 0`] = `
"// This is a comment
1 + 1;
"
`;

exports[`fjs simple-scope-breaking 0`] = `
"const foo = {
  meh: 'reasonably long value',
  meh2: 'this one will make things wrap, hopefully...',
};
"
`;

exports[`fjs string-escapes 0`] = `
"let x = '\\\\\\\\\\\\\\'';
"
`;

exports[`fjs variable-declarations 0`] = `
"var foo, bar;
var baz;
let buz = 1;
const qux = { one: 1, two: 2 };
"
`;

exports[`fjs works 0`] = `
"1 + 1;
"
`;
